src/lib/core/CHIPPersistentStorageDelegate.h

```diff
@@ -46,30 +46,20 @@ class DLL_EXPORT PersistentStorageDelegate
      *   Caller is responsible to take care of any special formatting needs (e.g. byte
      *   order, null terminators, consistency checks or versioning).
      *
-     *   This API allows for determining the size of a stored value. Whenever
-     *   the passed `size` is smaller than needed and the key exists in storage, the error
-     *   CHIP_ERROR_BUFFER_TOO_SMALL will be given, and the `size` will be updated to the
-     *   size of the stored value. It is legal to use `nullptr` for `buffer` if `size` is 0.
-     *
      *   If a key is found and the `buffer`'s `size` is large enough, then the value will
      *   be copied to `buffer` and `size` will be updated to the actual size used.
      *
-     *   The easiest way to determine if a key exists (and the value's size if so) is to pass
-     *   `size` of 0, which is always valid to do, and will return CHIP_ERROR_BUFFER_TOO_SMALL
-     *   if the key exists and CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND if the
-     *   key is not found.
+     *   Whenever the passed `size` is smaller than needed and the key exists in storage,
+     *   the error CHIP_ERROR_BUFFER_TOO_SMALL will be given. The `buffer` will be filled
+     *   up to`size`. It is legal to use `nullptr` for `buffer` if `size` is 0.
```

Comment:
```suggestion
     *   Whenever the passed `size` is smaller than the value for which the key exists in storage,
     *   CHIP_ERROR_BUFFER_TOO_SMALL will be given. With either CHIP_ERROR_BUFFER_TOO_SMALL
     *   or CHIP_NO_ERROR return values, the `buffer` will be filled with up to`size` bytes of the stored value.
     *   It is legal to use `nullptr` for `buffer` if `size` is 0.
```

Reply:
What is the point of passing size=0 if the API can't return to us the actual size?  This doesn't make sense.

Also, it would be nice to be able to learn the size.  Why are we introducing this limitation?

Reply:
It is to determine if a key is present

Reply:
>What is the point of passing size=0 if the API can't return to us the actual size? This doesn't make sense.

This allow us to know if a key exist

>Also, it would be nice to be able to learn the size. Why are we introducing this limitation?

This is addressed https://github.com/project-chip/connectedhomeip/pull/20164#issuecomment-1171492724